homeassistant/components/motionblinds_ble/sensor.py

```diff
@@ -0,0 +1,221 @@
+"""Sensor entities for the Motionblinds BLE integration."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+import logging
+from math import ceil
+
+from motionblindsble.const import (
+    MotionBlindType,
+    MotionCalibrationType,
+    MotionConnectionType,
+)
+from motionblindsble.device import MotionDevice
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE, EntityCategory
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import (
+    ATTR_BATTERY,
+    ATTR_CALIBRATION,
+    ATTR_CONNECTION,
+    ATTR_SIGNAL_STRENGTH,
+    CONF_MAC_CODE,
+    DOMAIN,
+    ICON_CALIBRATION,
+    ICON_CONNECTION,
+)
+from .entity import MotionblindsBLEEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+PARALLEL_UPDATES = 0
+
+
+@dataclass(frozen=True, kw_only=True)
+class MotionblindsBLESensorEntityDescription(SensorEntityDescription):
+    """Entity description of a sensor entity with native_value attribute."""
+
+    native_value: str | None = None
+
+
+SENSOR_TYPES: dict[str, MotionblindsBLESensorEntityDescription] = {
+    ATTR_BATTERY: MotionblindsBLESensorEntityDescription(
+        key=ATTR_BATTERY,
+        translation_key=ATTR_BATTERY,
+        native_unit_of_measurement=PERCENTAGE,
+        device_class=SensorDeviceClass.BATTERY,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    ATTR_CONNECTION: MotionblindsBLESensorEntityDescription(
+        key=ATTR_CONNECTION,
+        translation_key=ATTR_CONNECTION,
+        icon=ICON_CONNECTION,
+        device_class=SensorDeviceClass.ENUM,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        options=[connection_type.value for connection_type in MotionConnectionType],
+        native_value=MotionConnectionType.DISCONNECTED.value,
+    ),
+    ATTR_CALIBRATION: MotionblindsBLESensorEntityDescription(
+        key=ATTR_CALIBRATION,
+        translation_key=ATTR_CALIBRATION,
+        icon=ICON_CALIBRATION,
+        device_class=SensorDeviceClass.ENUM,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    ATTR_SIGNAL_STRENGTH: MotionblindsBLESensorEntityDescription(
+        key=ATTR_SIGNAL_STRENGTH,
+        translation_key=ATTR_SIGNAL_STRENGTH,
+        device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        native_unit_of_measurement="dBm",
+    ),
```

Comment:
Let device class handle the name for this one

Reply:
So should I only remove translation_key in that case and remove it from strings.json?

Reply:
yes

Reply:
Thanks done